devices/temp_nix.go

```diff
@@ -42,21 +42,26 @@ func getTemps(temps map[string]int) map[string]error {
 	for _, disk := range block.Disks {
 		dev, err := smart.Open("/dev/" + disk.Name)
 		if err != nil {
-			log.Println(err)
 			continue
 		}
 		defer dev.Close()
 
 		switch sm := dev.(type) {
 		case *smart.SataDevice:
 			data, _ := sm.ReadSMARTData()
-			for _, attr := range data.Attrs {
-				if attr.Id == 194 {
-					temps[disk.Name+"_"+disk.Model] = int(attr.Value)
-				}
+			if err != nil {
+				log.Print("error getting smart data")
+				return nil
+			}
+			if attr, ok := data.Attrs[194]; ok {
+				temps[disk.Name+"_"+disk.Model] = int(attr.Value)
```

Comment:
And here you might want to use `ParseAsTemperature()` or `ParseAsTemperatureRange()` here.

It is not clear to me how to handle situations when different drives format temperature in a different way. The temperature API might be adjusted.

Reply:
@anatol those new methods are quite handy, making use of them on b89c1dd748b8ee0913f99cdb16079e0515ae7aba